pandas/_libs/tslibs/timedeltas.pyx

```diff
@@ -241,24 +245,14 @@ cdef object ensure_td64ns(object ts):
         str unitstr
 
     td64_unit = get_datetime64_unit(ts)
-    if (
-        td64_unit != NPY_DATETIMEUNIT.NPY_FR_ns
-        and td64_unit != NPY_DATETIMEUNIT.NPY_FR_GENERIC
-    ):
```

Comment:
worth double-checking, but i think the current version here is slightly more performant bc it doesn't construct the tuple

Reply:
I haven't run the benchmarks yet, but happy to revert to the earlier version for now.